### PR TITLE
Updated python docs/examples for browsers section

### PR DESCRIPTION
**Pull request by pmartinez1**
### **User description**
Moved or added examples in python docs for browsers (for all missing examples)

**Thanks for contributing to the Selenium site and documentation!**
**A PR well described will help maintainers to review and merge it quickly**

Before submitting your PR, please check our [contributing](https://selenium.dev/documentation/en/contributing/) guidelines.
Avoid large PRs, and help reviewers by making them as simple and short as possible.

### Description

Added and moved all relevant python code examples for the browsers section.

### Motivation and Context

It will allow python users to see examples of working code so they can more easily and efficiently work with selenium.

### Types of changes

- [ x] Change to the site (I have double-checked the Netlify deployment, and my changes look good)
- [ x] Code example added (and I also added the example to all translated languages)
- [ ] Improved translation
- [ ] Added new translation (and I also added a notice to each document missing translation)

### Checklist

- [ x] I have read the [**contributing**](https://selenium.dev/documentation/en/contributing/) document.
- [ x] I have used [hugo](https://gohugo.io/) to render the site/docs locally and I am sure it works.

___

### **PR Type**
Documentation, Enhancement

___

### **Description**
- Added new Python tests for Chrome, Edge, Firefox, Internet Explorer, and Safari browsers.
- Updated documentation to include new Python code examples for various browsers.
- Improved the visibility of Python examples in multiple languages.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Tests</strong></td><td><details><summary>5 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>test_chrome.py</strong><dd><code>Add Chrome browser tests for various features</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

examples/python/tests/browsers/test_chrome.py

<li>Added tests for network conditions, logs, permissions, and casting.<br>

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1854/files#diff-47561b87e40e35009b0ce3e88512579da1f96ffb911e86fd8dc9dee6708e7ba0">+48/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>test_edge.py</strong><dd><code>Add Edge browser tests for various features</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

examples/python/tests/browsers/test_edge.py

<li>Added tests for network conditions, logs, permissions, and casting.<br>

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1854/files#diff-503affe7beda841e0a1233e41de7b930960f073ce60ef7843e1044adc4968eac">+44/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>test_firefox.py</strong><dd><code>Add Firefox browser tests for various features</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

examples/python/tests/browsers/test_firefox.py

- Added tests for full-page screenshots, context, and profiles.

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1854/files#diff-0c150a20628e9efa5a2d236dd4cd1c861ecbfc8f25f32a2add0958f9155cccf0">+30/-2</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>test_internet_explorer.py</strong><dd><code>Add Internet Explorer browser tests for various features</code>&nbsp; </dd></summary>
<hr>

examples/python/tests/browsers/test_internet_explorer.py

- Added multiple tests for Internet Explorer options.

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1854/files#diff-925dd314230279d419af16343e81eee569833f10ecf0a20d2a070e9b917494d5">+78/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>test_safari.py</strong><dd><code>Add Safari browser test for Technology Preview</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

examples/python/tests/browsers/test_safari.py

- Added a test for Safari Technology Preview.

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1854/files#diff-f983ddc390b967d5b05a4318f70f27802784099e585ebee5702a7f482964e67d">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Documentation</strong></td><td><details><summary>20 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>chrome.en.md</strong><dd><code>Update Chrome documentation with Python examples</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

website_and_docs/content/documentation/webdriver/browsers/chrome.en.md

- Updated Python code examples for Chrome browser.

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1854/files#diff-eccc7d2ca21e26be42c8c903fa062012754d5ea90f001a432f4f9961e87b078c">+4/-4</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>chrome.ja.md</strong><dd><code>Update Chrome documentation with Python examples (Japanese)</code></dd></summary>
<hr>

website_and_docs/content/documentation/webdriver/browsers/chrome.ja.md

- Updated Python code examples for Chrome browser.

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1854/files#diff-85cef9cb445defdf9de42ecc0dbd533e2ef2b0b47bcd8fb2aecc09f2114b3238">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>chrome.pt-br.md</strong><dd><code>Update Chrome documentation with Python examples (Portuguese)</code></dd></summary>
<hr>

website_and_docs/content/documentation/webdriver/browsers/chrome.pt-br.md

- Updated Python code examples for Chrome browser.

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1854/files#diff-4d648f7aea25c7e23a591cdb153ae6a6857d909ab673354eafdf3c5a3b4c4aa7">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>chrome.zh-cn.md</strong><dd><code>Update Chrome documentation with Python examples (Chinese)</code></dd></summary>
<hr>

website_and_docs/content/documentation/webdriver/browsers/chrome.zh-cn.md

- Updated Python code examples for Chrome browser.

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1854/files#diff-1aff253f48970f96fb108fece2476c5e0d815aec68c4275a2746b2bad32a377f">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>edge.en.md</strong><dd><code>Update Edge documentation with Python examples</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

website_and_docs/content/documentation/webdriver/browsers/edge.en.md

- Updated Python code examples for Edge browser.

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1854/files#diff-eba585ea4c9d1d71d09ee69558012785b4310856fd2b562e49433c4157766b36">+4/-4</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>edge.ja.md</strong><dd><code>Update Edge documentation with Python examples (Japanese)</code></dd></summary>
<hr>

website_and_docs/content/documentation/webdriver/browsers/edge.ja.md

- Updated Python code examples for Edge browser.

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1854/files#diff-3e852f27c226e1c9c7225112069d43d93dea6a09b6257647d8aaea9cd20e984b">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>edge.pt-br.md</strong><dd><code>Update Edge documentation with Python examples (Portuguese)</code></dd></summary>
<hr>

website_and_docs/content/documentation/webdriver/browsers/edge.pt-br.md

- Updated Python code examples for Edge browser.

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1854/files#diff-6ff8b3969797826d7ffa95108a6aa14cd83e2bfe5839f931ab1136bbea61409c">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>edge.zh-cn.md</strong><dd><code>Update Edge documentation with Python examples (Chinese)</code>&nbsp; </dd></summary>
<hr>

website_and_docs/content/documentation/webdriver/browsers/edge.zh-cn.md

- Updated Python code examples for Edge browser.

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1854/files#diff-7909b00888234283f45940e23c22d33a7ce74bef4b4ae8385b034429c9a4aec8">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>firefox.en.md</strong><dd><code>Update Firefox documentation with Python examples</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

website_and_docs/content/documentation/webdriver/browsers/firefox.en.md

- Updated Python code examples for Firefox browser.

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1854/files#diff-b15295d3d29dd29126510a0f6512397f6bb5ec98c5eb17c33798ef7b532aa4cf">+4/-9</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>firefox.ja.md</strong><dd><code>Update Firefox documentation with Python examples (Japanese)</code></dd></summary>
<hr>

website_and_docs/content/documentation/webdriver/browsers/firefox.ja.md

- Updated Python code examples for Firefox browser.

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1854/files#diff-02b45056986e5b04a1d62e9a4d294f0209c87f8937167aa405dcb7eea33f7617">+4/-9</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>firefox.pt-br.md</strong><dd><code>Update Firefox documentation with Python examples (Portuguese)</code></dd></summary>
<hr>

website_and_docs/content/documentation/webdriver/browsers/firefox.pt-br.md

- Updated Python code examples for Firefox browser.

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1854/files#diff-a7846d2db0bfab86e7a78f751e043ce8fc58a4abe79a9798efedce6def25ea3b">+4/-9</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>firefox.zh-cn.md</strong><dd><code>Update Firefox documentation with Python examples (Chinese)</code></dd></summary>
<hr>

website_and_docs/content/documentation/webdriver/browsers/firefox.zh-cn.md

- Updated Python code examples for Firefox browser.

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1854/files#diff-44bf88c57d3d9f7aa384e74e0f3b5ea1479e7d42eba14f7ba16f3bffb70b27d7">+4/-9</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>internet_explorer.en.md</strong><dd><code>Update Internet Explorer documentation with Python examples</code></dd></summary>
<hr>

website_and_docs/content/documentation/webdriver/browsers/internet_explorer.en.md

- Updated Python code examples for Internet Explorer browser.

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1854/files#diff-b887e55fa380db36be93803cc07213e0bfc1b22d72bd64e9716799c59f9970ec">+14/-71</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>internet_explorer.ja.md</strong><dd><code>Update Internet Explorer documentation with Python examples (Japanese)</code></dd></summary>
<hr>

website_and_docs/content/documentation/webdriver/browsers/internet_explorer.ja.md

- Updated Python code examples for Internet Explorer browser.

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1854/files#diff-d2dae06f1fde6e06d1b014e3d830b15bb2e7b6c7eea3472e304acbed2d770a02">+14/-71</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>internet_explorer.pt-br.md</strong><dd><code>Update Internet Explorer documentation with Python examples </code><br><code>(Portuguese)</code></dd></summary>
<hr>

website_and_docs/content/documentation/webdriver/browsers/internet_explorer.pt-br.md

- Updated Python code examples for Internet Explorer browser.

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1854/files#diff-fe2d01308c167a177a9d3bbb5ab5d10e71e644ca3fe386c5936fd9de12346b03">+14/-78</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>internet_explorer.zh-cn.md</strong><dd><code>Update Internet Explorer documentation with Python examples (Chinese)</code></dd></summary>
<hr>

website_and_docs/content/documentation/webdriver/browsers/internet_explorer.zh-cn.md

- Updated Python code examples for Internet Explorer browser.

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1854/files#diff-492f3c9f55f8b46ec55057baee400031d06cc5744d6d630ea1f160d849e65d86">+14/-71</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>safari.en.md</strong><dd><code>Update Safari documentation with Python examples</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

website_and_docs/content/documentation/webdriver/browsers/safari.en.md

- Updated Python code examples for Safari browser.

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1854/files#diff-634ecb7644504a2d7735debd2017c7d9091a4d042dd336cea63c521ffb9d9737">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>safari.ja.md</strong><dd><code>Update Safari documentation with Python examples (Japanese)</code></dd></summary>
<hr>

website_and_docs/content/documentation/webdriver/browsers/safari.ja.md

- Updated Python code examples for Safari browser.

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1854/files#diff-3843e74951875d9f02381b4a032de51b438b32a454e95d14a195a2f161646aee">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>safari.pt-br.md</strong><dd><code>Update Safari documentation with Python examples (Portuguese)</code></dd></summary>
<hr>

website_and_docs/content/documentation/webdriver/browsers/safari.pt-br.md

- Updated Python code examples for Safari browser.

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1854/files#diff-5bef1737606138f9714d746b1385f5a2b3de0002ed856fabf3e3e4c57ffe8bc7">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>safari.zh-cn.md</strong><dd><code>Update Safari documentation with Python examples (Chinese)</code></dd></summary>
<hr>

website_and_docs/content/documentation/webdriver/browsers/safari.zh-cn.md

- Updated Python code examples for Safari browser.

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1854/files#diff-a7937d029700732a1a80db8aaab9b545b2a5e14015ff3875aedc43f4a239aca8">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Comment by pmartinez1** [comment]
Am still going to test locally with the github code blocks. Will close this comment when done.

File: website_and_docs/content/documentation/webdriver/browsers/chrome.en.md [hunk 1]
```diff
@@ -373,7 +373,7 @@ You can drive Chrome Cast devices, including sharing tabs
 {{< badge-code >}}
 {{< /tab >}}
 {{% tab header="Python" %}}
```
Comment (pmartinez1) [inline website_and_docs/content/documentation/webdriver/browsers/chrome.en.md:375]: Am still going to test locally with the github code blocks. Will close this comment when done.

**Comment by pmartinez1** [comment]
Hi @harsha509,
Do you have any suggestion on how to test in Windows? I have a mac. 3 python examples tests failed. I believed I was able to resolve one of the failures (the error was pretty clear as to what was going on), but it would be best to fully confirm.

**Comment by pmartinez1** [comment]
Same failures, looks like I should not have passed in the second argument "True" in add_argument (https://github.com/SeleniumHQ/seleniumhq.github.io/blob/247ac2eea97e8d6031a357e84bb17f01c3175021/examples/python/tests/browsers/test_internet_explorer.py#L119). Quite possibly the other two are failing because they are trying to use the same instance of IE? Relevant: https://www.selenium.dev/documentation/ie_driver_server/. I'll look into this more and find a way of running it myself.

**Comment by pmartinez1** [comment]
> Same failures, looks like I should not have passed in the second argument "True" in add_argument (
> 
> https://github.com/SeleniumHQ/seleniumhq.github.io/blob/247ac2eea97e8d6031a357e84bb17f01c3175021/examples/python/tests/browsers/test_internet_explorer.py#L119
> ). Quite possibly the other two are failing because they are trying to use the same instance of IE? Relevant: https://www.selenium.dev/documentation/ie_driver_server/. I'll look into this more and find a way of running it myself.

Resolved the failure due to the "true" argument being passed in. The other failures are due to the use of the force_create_process_api value being set. According to the docs: "Windows Registry HKLM_CURRENT_USER\\Software\\Microsoft\\Internet Explorer\\Main path should contain key TabProcGrowth with 0 value."

I think the solution could be to have the test with force_create_process_api set to true to be skipped, as it could fail, if the value of the key above isn't set to 0. I will push that update, please let me know if it is not acceptable.

**Comment by pmartinez1** [comment]
> > Same failures, looks like I should not have passed in the second argument "True" in add_argument (
> > https://github.com/SeleniumHQ/seleniumhq.github.io/blob/247ac2eea97e8d6031a357e84bb17f01c3175021/examples/python/tests/browsers/test_internet_explorer.py#L119
> > 
> > ). Quite possibly the other two are failing because they are trying to use the same instance of IE? Relevant: https://www.selenium.dev/documentation/ie_driver_server/. I'll look into this more and find a way of running it myself.
> 
> Resolved the failure due to the "true" argument being passed in. The other failures are due to the use of the force_create_process_api value being set. According to the docs: "Windows Registry HKLM_CURRENT_USER\Software\Microsoft\Internet Explorer\Main path should contain key TabProcGrowth with 0 value."
> 
> I think the solution could be to have the test with force_create_process_api set to true to be skipped, as it could fail, if the value of the key above isn't set to 0. I will push that update, please let me know if it is not acceptable.

Updated and ready for review.

**Comment by pmartinez1** [comment]
Hi @harsha509 ,

The tests that were failing before, passed on the last run. The tests that failed, were not touched by me and they were timeouts. Could the checks be run again?

Thanks.

**Comment by pmartinez1** [comment]
Could I get a review please?

File: examples/python/tests/browsers/test_edge.py [hunk 2]
```diff
@@ -122,3 +122,47 @@ def test_build_checks(log_path):
...
+    driver = webdriver.Edge()
+    
+    driver.get('https://www.selenium.dev/webs')
+    browser_logs = driver.get_log('browser')
+    
+    assert 'Failed to load' in browser_logs[0]['message']
+    driver.quit()
+
+def test_permissions():
+    driver = webdriver.Chrome()
```
Comment (harsha509) [inline examples/python/tests/browsers/test_edge.py]: webdriver.Edge()

**Review by harsha509 — CHANGES_REQUESTED** [review]

File: examples/python/tests/browsers/test_edge.py (hunk 2 above)
Comment (pmartinez1) [reply examples/python/tests/browsers/test_edge.py]: Good catch! Updated.

**Review by harsha509 — APPROVED** [review]

**Comment by pmartinez1** [comment]
Looks like this has been failing its checks consistently. Updating to a draft, so I can figure out the issue. I'll update this PR once I have seen them all pass consistently.

**Comment by VietND96** [comment]
Could you please resolve conflict in this PR?

**Comment by pmartinez1** [comment]
Closing this PR as someone else implemented the same stuff already.

**Pull request closed without merging**